crates/driver/src/infra/api/error.rs

```diff
@@ -0,0 +1,104 @@
+use {
+    crate::{
+        domain::competition::{self, quote, solution},
+        infra::api,
+    },
+    hyper::StatusCode,
+    serde::Serialize,
+};
+
+#[derive(Debug, Clone, Copy, Serialize)]
+#[serde(rename_all = "PascalCase")]
+enum Type {
+    QuotingFailed,
+    SolverFailed,
+    SolutionNotFound,
+    DeadlineExceeded,
+    SimulationFailed,
+    Unknown,
+    TransactionPublishingFailed,
+    InvalidAuctionId,
+    MissingSurplusFee,
+    LiquidityError,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Error {
+    error_type: Type,
+    description: &'static str,
+}
+
+impl From<Type> for (StatusCode, axum::Json<Error>) {
+    fn from(value: Type) -> Self {
+        let (description, status_code) = match value {
+            Type::QuotingFailed => ("No valid quote found", StatusCode::BAD_REQUEST),
+            Type::SolverFailed => (
+                "Solver engine returned an invalid response",
+                StatusCode::BAD_REQUEST,
+            ),
+            Type::SolutionNotFound => ("No solution found for given ID", StatusCode::BAD_REQUEST),
+            Type::DeadlineExceeded => ("Exceeded solution deadline", StatusCode::BAD_REQUEST),
+            Type::SimulationFailed => ("Solution simulation failed", StatusCode::BAD_REQUEST),
+            Type::Unknown => ("An unknown error occurred", StatusCode::BAD_REQUEST),
+            Type::TransactionPublishingFailed => (
+                "Failed to publish the settlement transaction",
+                StatusCode::BAD_REQUEST,
+            ),
```

Comment:
Like for `LiquidityError` error, this doesn't sound like a 4xx client error, but a 5xx server error.